hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -143,46 +126,33 @@ public List<String> getAllPartitionPaths() throws IOException {
    * @param partitionPath The absolute path of the partition to list
    */
   @Override
-  public FileStatus[] getAllFilesInPartition(Path partitionPath)
-      throws IOException {
-    if (isMetadataTableEnabled) {
-      try {
-        return fetchAllFilesInPartition(partitionPath);
-      } catch (Exception e) {
-        throw new HoodieMetadataException("Failed to retrieve files in partition " + partitionPath + " from metadata", e);
-      }
+  public FileStatus[] getAllFilesInPartition(Path partitionPath) throws IOException {
+    ValidationUtils.checkArgument(isMetadataTableInitialized);
+    try {
+      return fetchAllFilesInPartition(partitionPath);
+    } catch (Exception e) {
+      throw new HoodieMetadataException("Failed to retrieve files in partition " + partitionPath + " from metadata", e);
     }
-
-    FileSystemBackedTableMetadata fileSystemBackedTableMetadata =
```

Comment:
lets move the validation check to constructor for metadata initialized.

Reply:
I think we should do this change in a separate PR. It is independent of RLI.

Reply:
HoodieTableMetadata.create should not return an instance of HoodieBackedTableMetadata if MDT is not initialized.

Reply:
Agree, but this behavior change (not falling back to FS backed metadata) caused many tests to fail. I'm fully aligned with the change but it could have been done in a separate PR and checked in before RLI.